geometry/permutation_test.cpp

```diff
@@ -1,11 +1,12 @@
 ﻿
-#include "geometry/permutation.hpp"
-
 #include <vector>
 
 #include "geometry/frame.hpp"
 #include "geometry/identity.hpp"
 #include "geometry/orthogonal_map.hpp"
+// TODO(egg): this should be at the top, but there's a circular dependency due
+// to compatibility test.
```

Comment:
I'm not sure what this means, but should we have include guards to deal with this oddity?